libs/botan/Makefile

```diff
@@ -0,0 +1,89 @@
+#
+# Copyright (C) 2019 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=botan
+PKG_VERSION:=2.9.0
+PKG_RELEASE:=1
+PKG_MAINTAINER:=BangLang Huang <banglang.huang@foxmail.com>
+
+PKG_SOURCE:=$(PKG_VERSION)
+PKG_SOURCE_URL:=https://codeload.github.com/randombit/botan/tar.gz/
+PKG_HASH:=5cd344bf92e69caa60a5d9464b5743229a90e879afe9d72bb8f2f43d28c70b68
+
+PKG_CAT:=zcat
+PKG_LICENSE_FILE:=license.txt
+PKG_INSTALL:=1
+
+include $(INCLUDE_DIR)/package.mk
+
+define Package/botan/Default
+  SUBMENU:=SSL
+  TITLE:=Crypto and TLS for C++11
+  URL:=https://botan.randombit.net
+endef
+
+define Package/botan/Default/description
+  Botan (Japanese for peony) is a cryptography library written in C++11 and
+  released under the permissive Simplified BSD license.
+endef
+
+define Package/libbotan
+  $(call Package/botan/Default)
+  SECTION:=libs
+  CATEGORY:=Libraries
+  TITLE+= (library)
+  ABI_VERSION:=$(PKG_VERSION)-$(PKG_RELEASE)
+  DEPENDS:=+libstdcpp +libpthread
+endef
+
+define Package/libbotan/description
+  $(call Package/botan/Default/description)
+  This package contains the botan library.
+endef
+
+CONFIGURE_CMD = ./configure.py
+
+CONFIGURE_ARGS = \
+	--cpu="$(ARCH)" \
+	--cc-bin="$(TOOLCHAIN_DIR)/bin/$(TARGET_CXX)" \
+	--program-suffix="" \
+	--prefix=$(CONFIGURE_PREFIX) \
+	--exec-prefix=$(CONFIGURE_PREFIX) \
+	--bindir=$(CONFIGURE_PREFIX)/bin \
+	--sbindir=$(CONFIGURE_PREFIX)/sbin \
+	--libexecdir=$(CONFIGURE_PREFIX)/lib \
+	--sysconfdir=/etc \
+	--datadir=$(CONFIGURE_PREFIX)/share \
+	--localstatedir=/var \
+	--mandir=$(CONFIGURE_PREFIX)/man \
+	--infodir=$(CONFIGURE_PREFIX)/info \
+	$(DISABLE_IPV6)
+
+CONFIGURE_ARGS += \
+	--cpu="$(ARCH)" \
+	--optimize-for-size
+
+TARGET_LDFLAGS += \
+	-Wl,--gc-sections \
```

Comment:
Changing to -Wl,--gc-sections,--as-needed can result in a slightly smaller size.